assets/js/add-reason.js

```diff
@@ -0,0 +1,230 @@
+        Promise.resolve([]));
+
+      const posts = answers.map((answer, index) => (
+        () => (
+          $.ajax({
+            type: 'POST',
+            url: `/reason-for-appealing/item-${index}`,
+            // eslint-disable-next-line id-blacklist
+            data: answer,
+            success: response => {
+              if (response.validationErrors.length > 0) {
+                self.handleValidationError(index, response.validationErrors);
+              } else if ($(`#items-${index}`).children().hasClass('form-group-error')) {
+                $(`#items-${index} .form-group`)
+                  .removeClass('form-group-error')
+                  .children()
+                  .remove('.error-message');
+              }
+            }
+          })
+        ))
+      );
+
+      return promiseSequence(posts)
+        .then(responses => {
+          const validationErrors = responses.filter(response => response.validationErrors);
+          const actualErrors = validationErrors.filter(error => error.validationErrors.length > 0);
+          if (actualErrors.length === 0) {
+            // eslint-disable-next-line no-invalid-this
+            this.submit();
+          } else {
+            self.handleErrorSummary(validationErrors);
+          }
+        });
+    });
+  }
+
+  handleErrorSummary(fieldErrors) {
+    const errorSummaryList = fieldErrors.map((fieldError, index) => (
+      // eslint-disable-next-line arrow-body-style
+      fieldError.validationErrors.map(validationError => ({
+        id: `items-${index}`,
+        message: validationError.errors[0]
+      }))
+    ));
+    const summary = this.buildErrorSummary(flatten(errorSummaryList));
+    $('.error-summary').remove();
+    $('.column-two-thirds').prepend(summary.val);
+  }
+
+  handleValidationError(index, validationErrors) {
+    const whatYouDisagreeWith = validationErrors[0];
+    const reasonForAppealing = validationErrors[1];
+    const errorTextbox = this.buildWhatYouDisagreeWithField(
+      whatYouDisagreeWith.errors,
+      whatYouDisagreeWith.value
+    );
+    const errorTextArea = this.buildReasonForAppealingField(
+      reasonForAppealing.errors,
+      reasonForAppealing.value
+    );
+    $(`#items-${index}`).empty()
+      .append(errorTextbox.val)
+      .append(errorTextArea.val);
+  }
+
+  buildAnswers(index) {
+    const whatYouDisagreeWith = $(`#items-${index} #item\\.whatYouDisagreeWith`).val();
+    const reasonForAppealing = $(`#items-${index} #item\\.reasonForAppealing`).val();
+    return {
+      'item.whatYouDisagreeWith': whatYouDisagreeWith,
+      'item.reasonForAppealing': reasonForAppealing
+    };
+  }
+
+  addAnother() {
+    $('.add-another-add-link').click(event => {
+      event.preventDefault();
+      this.counter += 1;
+      this.addFields();
+    });
+  }
+
+  static startAddReason() {
+    return (window.location.pathname === '/reason-for-appealing');
```

Comment:
I know I wrote this line 😊 but anyway, maybe it's better to put the url in a constant.

Reply:
Actually yeah, i should import it from paths.js